host/CMakeLists.txt

```diff
@@ -388,9 +388,17 @@ if (OE_SGX AND UNIX)
   # for ocall stack-stitching by using the -fno-omit-frame-pointer flag.
   # It is compiled with -O2 flag to retain the same generated assembly
   # code in both debug and release builds.
-  set_source_files_properties(
-    oehost sgx/linux/enter.c PROPERTIES COMPILE_FLAGS
-                                        "-O2 -fno-omit-frame-pointer")
+  # Don't add -Wno-frame-address for clang-10 as it's not supported
```

Comment:
Why is -Wno-frame-address needed?

Reply:
We added "-Wno-frame-address" because for clang-11, because we use `__builtin_frame_address(1)` in [enclave/core/sgx/calls.c](https://github.com/openenclave/openenclave/blob/99ce2b87a173c07e66b64db43fd2ba0a65ef53e5/enclave/core/sgx/calls.c#L1287) which triggers `-Wframe-address` which returns a warning whenever `__builtin_frame_address()` is called with an argument greater than 0. If I recall correctly, even though it is a warning, it caused the build to fail.

Reply:
@anakrish  - Is this a concern?

Reply:
`enclave/core/sgx/calls.c` is an enclave file. The source file properties set on the host file `sgx/linux/enter.c` will not affect warnings in the former file - the compiler sees the two files separately.

However, I do see calls to `__builtin_frame_address(1)` in host/linux/sgx/enter.c
https://github.com/openenclave/openenclave/blob/2003a7bc1f91343ab120571cbaadfc271cef7a7a/host/sgx/linux/enter.c#L133

The comment could be changed to reflect why we are adding -Wno-frame-address for GCC, and not for clang.

Reply:
Ah yes, there are equivalent changes within `host/` and `enclave/` so I should've referred to the right one. Good catch that `-Wno-frame-address` is actually a GCC option, not Clang. I will fix that.